File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/2_organisasi_regional/logic/python/membership_movement/2_leave_logic/regional_leave_handler.py
import random

class RegionalLeaveHandler:
    """
    Handles the logic for an AI country to leave a regional organization.
    Exit probability is lower than global (10% vs 15%) to show regional commitment.
    """
    def __init__(self, exit_probability=0.10):
        self.exit_probability = exit_probability

    def evaluate_leave(self, country_name, decision_weight):
        """
        Returns True if the country decides to leave.
        """
        # Regional blocks are harder to leave unless relations are catastrophic
        if decision_weight < 15:
            if random.random() < self.exit_probability:
                return True
        return False

File: app/frontend/src/app/game/components/2_navigasi_menu/2_navigasi_bawah/5_geopolitik/2_organisasi_internasional/2_organisasi_regional/logic/python/membership_movement/2_leave_logic/RegionalLeaveHandler.ts
/**
 * RegionalLeaveHandler TS Bridge
 */
export class RegionalLeaveHandler {
    private exitProbability: number;

    constructor(exitProbability = 0.10) {
        this.exitProbability = exitProbability;
    }

    public evaluateLeave(countryName: string, decisionWeight: number): boolean {
        // Regional blocks are harder to leave unless relations are catastrophic
        if (decisionWeight < 15) {
            return Math.random() < this.exitProbability;
        }
        return false;
    }
}
